packages/virtualized-lists/Lists/VirtualizedList.js

```diff
@@ -1121,6 +1184,15 @@ class VirtualizedList extends StateSafePureComponent<Props, State> {
     if (hiPriInProgress) {
       this._hiPriInProgress = false;
     }
+
+    // Make sure to cancel any pending updates if maintainVisibleContentPositionAdjustment
+    // changed since they are now invalid.
+    if (
+      prevState.maintainVisibleContentPositionAdjustment !==
+      this.state.maintainVisibleContentPositionAdjustment
+    ) {
+      this._updateCellsToRenderBatcher.dispose({abort: true});
+    }
```

Comment:
I will double check if this is still needed now that we adjust cellsAroundViewport.

Reply:
Will also check the code that you mentioned for handling stale scroll position see if it is possible to leverage it instead of skipping calls.

Reply:
This code seems to be still needed, or at least some way to skip updates while we expect a scroll position adjustment since it uses `_scrollMetrics` which will be out of sync with native cause cause a wrong window to be rendered.

Reply:
I think folding into `_adjustCellsAroundViewport` will be more complex since there are some other functions like for example `_updateViewableItems` that should not be called with invalid scroll metrics, and it is also called from setState update which I think makes it impossible to use this.state so we'd need to pass `pendingScrollUpdateCount` but at this point might as well just skip calling the function.

One option would be to move the check to `_updateCellsToRender` which would avoid having to check in `_scheduleCellsToRenderUpdate` and cancel pending batches. The check in `_maybeCallOnEdgeReached` is also needed since it is called from many places like `onLayout` which will happen before we get adjusted scroll position.